File: Frontend/projects/edu-sharing-api/src/lib/api/models/open-id-registration-result.ts
/* tslint:disable */
/* eslint-disable */
import { LtiToolConfiguration } from './lti-tool-configuration';
export interface OpenIdRegistrationResult {
    application_type?: string;
    client_id?: string;
    client_name?: string;
    grant_types?: Array<string>;
    'https://purl.imsglobal.org/spec/lti-tool-configuration'?: LtiToolConfiguration;
    initiate_login_uri?: string;
    jwks_uri?: string;
    logo_uri?: string;
    redirect_uris?: Array<string>;
    response_types?: Array<string>;
    scope?: string;
    token_endpoint_auth_method?: string;
}

File: Backend/services/core/src/main/java/org/edu_sharing/restservices/ltiplatform/v13/model/OpenIdRegistrationResult.java
package org.edu_sharing.restservices.ltiplatform.v13.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
public class OpenIdRegistrationResult {
    private String client_id;
    private List<String> response_types;
    private String jwks_uri;
    private String initiate_login_uri;
    private List<String> grant_types;
    private List<String> redirect_uris;
    private String application_type;
    private String token_endpoint_auth_method;
    private String client_name;
    private String logo_uri;
    private String scope;

    @JsonProperty("https://purl.imsglobal.org/spec/lti-tool-configuration")
    private LTIToolConfiguration ltiToolConfiguration;

    @Data
    public static class LTIToolConfiguration {
        private String version;
        private String deployment_id;
        private String target_link_uri;
        private String domain;
        private String description;
        private List<String> claims;
    }
}
